app/src/main/java/com/microsoft/office365/msgraphsnippetapp/SignInActivity.java

```diff
@@ -114,17 +141,94 @@ private void warnBadClient() {
         validateOrganizationArgs();
-        mAuthenticationManager.connect(this);
+        connect();
+    }
+
+    private void connect() {
+
+        // The sample app is having the PII enable setting on the MainActivity. Ideally, app should decide to enable Pii or not,
+        // if it's enabled, it should be  the setting when the application is onCreate.
+        if (mEnablePiiLogging) {
+            Logger.getInstance().setEnablePII(true);
+        } else {
+            Logger.getInstance().setEnablePII(false);
+        }
+
+        /* Attempt to get a user and acquireTokenSilent
+         * If this fails we do an interactive request
+         */
+        List<IAccount> users = null;
+
+        try {
+            users = mAuthenticationManager.getPublicClient().getAccounts();
+
+            if (users != null && users.size() == 1) {
+                /* We have 1 user */
+                mUser = users.get(0);
+                mAuthenticationManager.callAcquireTokenSilent(
+                        mUser,
+                        true,
+                        this);
+            } else {
+                /* We have no user */
+
+                /* Let's do an interactive request */
+                mAuthenticationManager.callAcquireToken(
+                        this);
+            }
+        }
+
+        catch (IndexOutOfBoundsException e) {
+            Log.d(TAG, "User at this position does not exist: " + e.toString());
+            Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show();
+
+
+        } catch (IllegalStateException e) {
+            Log.d(TAG, "MSAL Exception Generated: " + e.toString());
+            Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show();
+
+
+        } catch (Exception e) {
+            Toast.makeText(this, e.getMessage(), Toast.LENGTH_SHORT).show();
+        }
     }
 
     private void validateOrganizationArgs() throws IllegalArgumentException {
         UUID.fromString(ServiceConstants.CLIENT_ID);
-        URI.create(ServiceConstants.REDIRECT_URI);
     }
 
     private void start() {
         Intent appLaunch = new Intent(this, SnippetListActivity.class);
         startActivity(appLaunch);
     }
 
-}
+    public IAccount mUser;
+
+    @Override
+    public void onError(Exception exception) {
+        Toast.makeText(this, exception.getMessage(), Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void onCancel() {
+        Toast.makeText(this, "User cancelled the flow.", Toast.LENGTH_SHORT).show();
+    }
+    /**
+     * Handles redirect response from https://login.microsoftonline.com/common and
+     * notifies the MSAL library that the user has completed the authentication
+     * dialog
+     * @param requestCode
+     * @param resultCode
+     * @param data
+     */
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (mAuthenticationManager
```

Comment:
To improve readability make the expression in the if statement into a variable

```
PublicClient client = mAuthenticationManager.getPublicClient()

if (client != null) {
...
}
```

Reply:
Resolved. Thanks